include/benchmark/benchmark_api.h

```diff
@@ -693,8 +757,59 @@ class Benchmark {
   Benchmark& operator=(Benchmark const&);
 };
 
+// Incomplete iterator-like type with dummy value type so that benchmark::State
+// can support iteration with a range-based for loop. See comments on
+// benchmark::begin() / benchmark::end() concerning usage.
+class StateIterator {
+ public:
+  struct Value {
+    ~Value() {}  // Non-trivial destructor to avoid unused variable warnings
```

Comment:
This seems like a bad way to suppress unused value warnings. Having a non-trivial destructor could, when it's not optimized away, potentially undermine the entire point of this; which is writing the most efficient benchmark possible.

Using `struct BENCHMARK_UNUSED Value` suppresses the warning for GCC, we should use that instead.